ui/app/components/container-image-tag/index.ts

```diff
@@ -0,0 +1,57 @@
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+import ApiService from 'waypoint/services/api';
+import { StatusReport } from 'waypoint-pb';
+import { tracked } from '@glimmer/tracking';
+
+interface DockerImageBadgeArgs {
+  statusReport: StatusReport.AsObject;
+}
+
+interface StateJsonConfig {
+  Image: string;
+}
+
+export default class DockerImageBadge extends Component<DockerImageBadgeArgs> {
+  @service api!: ApiService;
+  @tracked image?: string;
+  @tracked tag?: string;
+  imageFromJson?: string;
+
+  constructor(owner: unknown, args: DockerImageBadgeArgs) {
+    super(owner, args);
+
+    this.parseImageAndTag();
+  }
+
+  findImageKey(obj: Record<string, unknown>): void {
+    if (typeof obj !== 'object') {
+      return;
+    }
+
+    for (let k in obj) {
+      if (k === 'Config') {
+        let config = obj[k] as StateJsonConfig;
+        this.imageFromJson = config.Image;
+      }
+
+      if (!obj || typeof obj[k] === 'object') {
+        this.findImageKey(obj[k] as Record<string, unknown>);
+      }
+    }
+    return;
+  }
+
+  parseImageAndTag(): void {
+    if (!this.args.statusReport || !this.args.statusReport.resourcesList) {
+      return;
+    }
+
+    let container = this.args.statusReport.resourcesList.find((r) => r.type === 'container');
```

Comment:
I’m realizing that this method of digging up the image reference isn’t general enough. On the resource detail page (which I wrote) it only works for Kubernetes pods, and here it only works for Docker containers. We should collaborate a general approach to this.

Reply:
I tried traversing everything in `statusReport.resourcesList` in search of `image` fields, this does get us what we want, but also some stuff we don’t want.

![image](https://user-images.githubusercontent.com/34030/135614968-671cadb4-91f8-467c-b600-795ec66c5302.png)

Here’s roughly what that approach would look like: https://github.com/hashicorp/waypoint/commit/cfbef23817ec7e6eabd9e30c9c8ec6ac5f93da98